iox_catalog/migrations/20230831090842_check_sort_key_sync.sql

```diff
@@ -0,0 +1,11 @@
+-- sanity check for https://github.com/influxdata/influxdb_iox/issues/8570
+DO $$
+DECLARE
+    rem integer;
+BEGIN
+    SELECT COUNT(*) INTO STRICT rem FROM partition WHERE cardinality(sort_key_ids) != cardinality(sort_key);
```

Comment:
```suggestion
    SELECT COUNT(*) INTO STRICT rem FROM partition WHERE (cardinality(sort_key_ids) != cardinality(sort_key)) OR sort_key_ids IS NULL;
```

Form my verification yesterday, I needed to add `OR sort_key_ids IS NULL` to see what was not migrated, too.  I think with this, we will capture everything did not go thru.

I did not run the query above with OR added in yet. You may want to rerun it locally first

Reply:
Not required. They cannot be NULL as per https://github.com/influxdata/influxdb_iox/blob/main/iox_catalog/migrations/20230822120825_sort_key_ids_not_null_take_2.sql .